### PR TITLE
render BackdropFilter layers directly to DisplayListBuilder

### PR DESCRIPTION
**Pull request by flar**
The Impeller code to render backdrop filters was hampered by the fact that we rendered them through calls to a Skia recorder which obfuscated the type of filter. This change will render the filters/saveLayers directly to the underlying DisplayListBuilder when using Impeller.

### CODE REVIEW

**Comment by flar** [comment]
I've found at least one bug while writing a more specific test case. Working on expanding the test cases to check for more bugs.

(For those wondering, one if the negative value tests in the abs_finite() method didn't negate the value upon discovery.)

**Review by bdero — APPROVED** [review]

**Review by knopp — APPROVED** [review]
The diff part looks good to me.

**Comment by flar** [comment]
I filed https://github.com/flutter/flutter/issues/106587 to improve the bounds under a perspective transform. Until then, the bounds calculations in DlImageFilter should be at least as "correct" as those computed by the associated Skia objects.

File: display_list/display_list_image_filter.h [hunk 1]
```diff
@@ -102,8 +102,9 @@ class DlImageFilter
   // (untransformed) coordinate space.
   //
   // The output bounds parameter must be supplied and the method will either
-  // return a pointer to it with the result filled in, or it will return a
-  // nullptr if it cannot determine the results.
+  // return a pointer to it with the result filled in, or it will set it to
+  // a copy of the |output_bounds| and return a nullptr if it cannot
```
[the remark this answers is not in the document]
Comment (flar) [reply display_list/display_list_image_filter.h]: Fixed

File: display_list/display_list_image_filter.h [hunk 2]
```diff
@@ -115,13 +116,49 @@ class DlImageFilter
   // Blur filter will expand based on the ctm).
   //
   // The output bounds parameter must be supplied and the method will either
-  // return a pointer to it with the result filled in, or it will return a
-  // nullptr if it cannot determine the results.
+  // return a pointer to it with the result filled in, or it will set it to
+  // a copy of the |output_bounds| and return a nullptr if it cannot
```
[the remark this answers is not in the document]
Comment (flar) [reply display_list/display_list_image_filter.h]: Fixed

**Comment by flar** [comment]
Now that the bounds computations are updated (fixed), this PR is no longer a WIP and is ready for final review.

**Comment by flar** [comment]
Now fixes https://github.com/flutter/flutter/issues/106587

**Pull request merged**